connection_test.go

```diff
@@ -575,156 +554,107 @@ func executeQueryAndConfirmMessage(db *sql.DB, query string, expectedErrorTable
 }
 
 func TestQueryArrowStreamError(t *testing.T) {
-	ctx := context.Background()
-	numrows := 50000 // approximately 10 ArrowBatch objects
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(ctx, *config)
-	if err != nil {
-		t.Error(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Error(err)
-	}
-
-	query := fmt.Sprintf(selectRandomGenerator, numrows)
-	sr := &snowflakeRestful{
-		FuncPostQuery:  postQueryTest,
-		TokenAccessor:  getSimpleTokenAccessor(),
-		RequestTimeout: 10,
-	}
-	sc.rest = sr
-	_, err = sc.QueryArrowStream(ctx, query)
-	if err == nil {
-		t.Error("should have raised an error")
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		sc := sct.sc
+		numrows := 50000 // approximately 10 ArrowBatch objects
+		query := fmt.Sprintf(selectRandomGenerator, numrows)
+		sr := &snowflakeRestful{
+			FuncPostQuery:    postQueryTest,
+			FuncCloseSession: closeSessionMock,
+			TokenAccessor:    getSimpleTokenAccessor(),
+			RequestTimeout:   10,
+		}
+		sc.rest = sr
+		_, err := sc.QueryArrowStream(sc.ctx, query)
+		if err == nil {
+			t.Error("should have raised an error")
+		}
 
-	sc.rest.FuncPostQuery = postQueryFail
-	_, err = sc.QueryArrowStream(ctx, query)
-	if err == nil {
-		t.Error("should have raised an error")
-	}
-	_, ok := err.(*SnowflakeError)
-	if !ok {
-		t.Fatalf("should be snowflake error. err: %v", err)
-	}
+		sc.rest.FuncPostQuery = postQueryFail
+		_, err = sc.QueryArrowStream(sc.ctx, query)
+		if err == nil {
+			t.Error("should have raised an error")
+		}
+		_, ok := err.(*SnowflakeError)
+		if !ok {
+			t.Fatalf("should be snowflake error. err: %v", err)
+		}
+	})
 }
 
 func TestExecContextError(t *testing.T) {
-	ctx := context.Background()
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(ctx, *config)
-	if err != nil {
-		t.Error(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Error(err)
-	}
-
-	sr := &snowflakeRestful{
-		FuncPostQuery:  postQueryTest,
-		TokenAccessor:  getSimpleTokenAccessor(),
-		RequestTimeout: 10,
-	}
-
-	sc.rest = sr
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		sc := sct.sc
+		ctx := sc.ctx
+		sr := &snowflakeRestful{
+			FuncPostQuery:  postQueryTest,
+			TokenAccessor:  getSimpleTokenAccessor(),
+			RequestTimeout: 10,
+		}
+		sc.rest = sr
 
-	_, err = sc.ExecContext(ctx, "SELECT 1", []driver.NamedValue{})
-	if err == nil {
-		t.Fatalf("should have raised an error")
-	}
+		_, err := sc.ExecContext(ctx, "SELECT 1", []driver.NamedValue{})
```

Comment:
Maybe we can introduce `mustExecContext` here?

Reply:
Not in this case. We need error here.